relayer-engine/src/storage/storage.ts

```diff
@@ -32,29 +32,70 @@ const CHECK_REQUEUED_WORKFLOWS_LOCK = "__isRequeueJobRunning";
 const CHECK_STALE_ACTIVE_WORKFLOWS_LOCK = "_isStaleWorkflowsJobRunning";
 const EMITTER_KEY = "__emitter";
 
+const CONSTANTS = {
+  READY_WORKFLOW_QUEUE,
+  ACTIVE_WORKFLOWS_QUEUE,
+  DEAD_LETTER_QUEUE,
+  DELAYED_WORKFLOWS_QUEUE,
+  EXECUTORS_HEARTBEAT_HASH,
+  STAGING_AREA_KEY,
+  CHECK_REQUEUED_WORKFLOWS_LOCK,
+  CHECK_STALE_ACTIVE_WORKFLOWS_LOCK,
+  EMITTER_KEY
+}
+
 type SerializedWorkflowKeys = { [k in keyof Workflow]: string | number };
 
-export async function createStorage(
-  plugins: Plugin[],
-  config: CommonEnv,
-  nodeId: string,
-  logger?: Logger,
-): Promise<Storage> {
+type CreationArgs = {
+  plugins: Plugin[]
+  config: CommonEnv
+  nodeId: string
+  logger: Logger
+}
+
+function validateAndGetRedisConfig(config: CommonEnv) {
   const redisConfig = config as RedisConfig;
   if (!redisConfig.redisHost || !redisConfig.redisPort) {
     throw new EngineError(
       "Redis config values must be present if redis store type selected",
     );
   }
+  return redisConfig;
+}
+
+async function createRedisStorage({ plugins, config, nodeId, logger}: CreationArgs) {
+  const redisConfig = validateAndGetRedisConfig(config);
+  const { defaultWorkflowOptions, namespace } = config;
   return new Storage(
     await RedisWrapper.fromConfig(redisConfig),
     plugins,
-    config.defaultWorkflowOptions,
+    defaultWorkflowOptions,
     nodeId,
-    logger || getLogger(),
+    logger,
+    namespace
   );
 }
 
+function createInvalidStorage({ config: { storeType } } : CreationArgs) {
```

Comment:
What is this complexity with factories buying us?

Reply:
I will revert, I was thinking to normalize the creator interface exposing a factories' dictionary, even for the case when the storyType is not supported.